tsl/src/nodes/skip_scan/planner.c

```diff
@@ -422,51 +485,71 @@ get_distinct_var(PlannerInfo *root, IndexPath *index_path, SkipScanPath *skip_sc
 
 	RangeTblEntry *ht_rte = planner_rt_fetch(var->varno, root);
 	RangeTblEntry *chunk_rte = planner_rt_fetch(rel->relid, root);
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_rte->relid, false);
 
-	/* Check for hypertable */
-	if (!ts_is_hypertable(ht_rte->relid) || !bms_is_member(var->varno, rel->top_parent_relids))
-		return NULL;
+	/* Check for compressed chunk */
+	if (chunk != NULL && ts_chunk_contains_compressed_data(chunk))
+	{
+		skip_scan_path->is_compressed = true;
+		AttrNumber attno = ts_map_attno(ht_rte->relid, chunk_rte->relid, var->varattno);
 
-	Relation ht_rel = table_open(ht_rte->relid, AccessShareLock);
-	Relation chunk_rel = table_open(chunk_rte->relid, AccessShareLock);
-	bool found_wholerow;
-	TupleConversionMap *map =
-		convert_tuples_by_name(RelationGetDescr(chunk_rel), RelationGetDescr(ht_rel));
+		foreach (lc, rel->reltarget->exprs)
+		{
+			Expr *e = (Expr *) lfirst(lc);
+			Var *v = castNode(Var, e);
+			if (v->varattno == attno)
+				return v;
+		}
+
+		return NULL;
+	}
 
-	/* attno mapping necessary */
-	if (map)
+	/* Check for hypertable */
+	if (ts_is_hypertable(ht_rte->relid) && bms_is_member(var->varno, rel->top_parent_relids))
 	{
-		var = (Var *) map_variable_attnos((Node *) var,
-										  var->varno,
-										  0,
-										  map->attrMap,
-										  InvalidOid,
-										  &found_wholerow);
-
-		free_conversion_map(map);
-
-		/* If we found whole row here skipscan wouldn't be applicable
-		 * but this should have been caught already in previous checks */
-		Assert(!found_wholerow);
-		if (found_wholerow)
+		Relation ht_rel = table_open(ht_rte->relid, AccessShareLock);
+		Relation chunk_rel = table_open(chunk_rte->relid, AccessShareLock);
+		bool found_wholerow;
+		TupleConversionMap *map =
+			convert_tuples_by_name(RelationGetDescr(chunk_rel), RelationGetDescr(ht_rel));
+
+		/* attno mapping necessary */
+		if (map)
```

Comment:
Can we use this code for when the `chunk_rel` is a compressed chunk? The attno mapping should work in the same way I think.